File: client/src/main/java/au/com/codeka/warworlds/client/util/UrlFetcher.java
package au.com.codeka.warworlds.client.util;

import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.URL;

/**
 * Helper class to fetch the contents of a URL.
 */
public class UrlFetcher {
  public static InputStream fetchStream(String urlString) throws IOException {
    URL url = new URL(urlString);
    HttpURLConnection conn = (HttpURLConnection) url.openConnection();
    conn.addRequestProperty("User-Agent", "wwmmo/" + Version.string());
    if (conn.getResponseCode() == 200) {
      return conn.getInputStream();
    }
    return null;
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/util/UrlFetcher.kt
package au.com.codeka.warworlds.client.util

import java.io.IOException
import java.io.InputStream
import java.net.HttpURLConnection
import java.net.URL

/**
 * Helper class to fetch the contents of a URL.
 */
object UrlFetcher {
  @JvmStatic
  @Throws(IOException::class)
  fun fetchStream(urlString: String?): InputStream? {
    val url = URL(urlString)
    val conn = url.openConnection() as HttpURLConnection
    conn.addRequestProperty("User-Agent", "wwmmo/" + Version.string())
    return if (conn.responseCode == 200) {
      conn.inputStream
    } else null
  }
}